flux-middle/src/fhir.rs

```diff
@@ -145,9 +145,10 @@ pub enum Ty {
 pub struct ArrayLen;
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
-pub enum RefKind {
+pub enum WeakKind {
```

Comment:
Oh, I think I explained myself wrong. We should not modify `RefKind`, but create a new type `WeakKind` which is only used in `PathsTree`